server/region_handler.go

```diff
@@ -40,13 +44,74 @@ const (
 	pDBName    = "db"
 	pTableName = "table"
 	pRegionID  = "regionID"
+	pRecordID  = "recordID"
+	pStartTS   = "startTS"
 )
 
 const (
 	headerContentType = "Content-Type"
 	contentTypeJSON   = "application/json"
 )
 
+type kvStore interface {
+	GetRegionCache() *tikv.RegionCache
+	SendReq(bo *tikv.Backoffer, req *tikvrpc.Request, regionID tikv.RegionVerID, timeout time.Duration) (*tikvrpc.Response, error)
+}
+
+type regionHandlerTool struct {
+	bo          *tikv.Backoffer
+	regionCache *tikv.RegionCache
+	store       kvStore
+}
+
+// RegionHandler is the common field for http region handler. It contains
+// some common functions for all handlers.
+type regionHandler struct {
+	*regionHandlerTool
+}
+
+// tableRegionsHandler is the handler for list table's regions.
+type tableRegionsHandler struct {
+	*regionHandler
+}
+
+// MvccTxnHandler is the handler for txn debugger
+type mvccTxnHandler struct {
+	*regionHandler
+	op string
+}
+
+const (
+	opMvccGetByKey = "key"
+	opMvccGetByTxn = "txn"
+)
+
+// prepare checks and prepares for region request. It returns
+// regionHandlerTool on success while return an err on any
+// error happens.
+func (s *Server) newRegionHandler() (hanler *regionHandler, err error) {
+	// check store
+	if s.cfg.Store != "tikv" {
+		err = fmt.Errorf("only store tikv support,current store:%s", s.cfg.Store)
+		return
+	}
+
+	store := s.driver.(*TiDBDriver).store
```

Comment:
Frankly speaking, those kind of logic is vulnerable, if the implementation change one day,  we got a panic here.